core/src/main/scala/org/apache/spark/deploy/master/Master.scala

```diff
@@ -566,10 +566,15 @@ private[deploy] class Master(
     // Update application state if executors are accepted and RUNNING
     apps.foreach(appInfo => {
       val app = idToApp(appInfo.id)
-      if(app.executors.filter(_._2.state != ExecutorState.RUNNING).isEmpty) {
-        app.state = ApplicationState.RUNNING
-        logInfo(s"Application :: ${app.id} status updated to RUNNING state")
-      }})
+      apps.foreach(f = appInfo => {
+        val app = idToApp(appInfo.id)
+        if (app.executors.size > 0 &&
```

Comment:
It might be more clear to write this condition like this
```scala
if (app.executors.nonEmpty && app.executors.forall(_._2.state == ExecutorState.RUNNING))
```